### PR TITLE
Fix #4691

### LINKED ISSUE #4691: Redundant checks in KeyedCollection<,> classes
In methods [Contains](https://github.com/dotnet/corefx/blob/49b8925bcf2bd1bd742915fe2f673b153dc74942/src/System.ObjectModel/src/System/Collections/ObjectModel/KeyedCollection.cs#L91) and [Remove](https://github.com/dotnet/corefx/blob/49b8925bcf2bd1bd742915fe2f673b153dc74942/src/System.ObjectModel/src/System/Collections/ObjectModel/KeyedCollection.cs#L131) we have a checks
```cs
            if (key == null)
            {
                throw new ArgumentNullException("key");
            }

            // …

            if (key != null) // <-- this "if" is redundant because we checked key == null above
            {
                // …
            }
```
Should this be fixed?

### PR DESCRIPTION
**Pull request by hughbe**
Removes redundant null checks in KeyedCollection

### CODE REVIEW

File: src/System.ObjectModel/src/System/Collections/ObjectModel/KeyedCollection.cs [hunk 1]
```diff
@@ -87,13 +87,10 @@ public bool Contains(TKey key)
             {
                 return _dict.ContainsKey(key);
             }
-
-            if (key != null)
+
```
Comment (justinvp) [inline src/System.ObjectModel/src/System/Collections/ObjectModel/KeyedCollection.cs]: nit: trailing whitespace

File: src/System.ObjectModel/src/System/Collections/ObjectModel/KeyedCollection.cs [hunk 2]
```diff
@@ -127,16 +124,13 @@ public bool Remove(TKey key)
                 TItem item;
                 return _dict.TryGetValue(key, out item) && Remove(item);
             }
-
-            if (key != null)
+
```
Comment (justinvp) [inline src/System.ObjectModel/src/System/Collections/ObjectModel/KeyedCollection.cs]: nit: trailing whitespace

**Comment by stephentoub** [comment]
LGTM once Justin's nits are addressed. I'd also be good to change the commit message to something more descriptive, so someone reading history doesn't need to crosscheck with issues.

**Comment by hughbe** [comment]
@stephentoub  all done

**Pull request merged**